tests/test_full_nlp.py

```diff
@@ -0,0 +1,198 @@
+        original_output.detach().numpy()
+        if isinstance(original_output, torch.Tensor) else original_output.logits.detach().numpy())
+
+
+@device('cpu', 'gpu')
+@pytest.mark.parametrize('model_type,algorithms,save_format', [('tinybert_hf', [GatedLinearUnits()], 'onnx'),
+                                                               ('simpletransformer', [], 'torchscript')])
+def test_full_nlp_pipeline(model_type, algorithms, save_format, tiny_bert_tokenizer, tmp_path, request, device):
+    device = get_device(device)
+
+    tiny_bert_model = None
+    if model_type == 'tinybert_hf':
+        tiny_bert_model = request.getfixturevalue('tiny_bert_model')
+
+    # pretraining
+    if model_type == 'tinybert_hf':
+        assert tiny_bert_model is not None
+        pretraining_metrics = [
+            LanguageCrossEntropy(ignore_index=-100, vocab_size=tiny_bert_tokenizer.vocab_size),
+            MaskedAccuracy(ignore_index=-100)
+        ]
+        pretraining_model = HuggingFaceModel(tiny_bert_model,
+                                             tiny_bert_tokenizer,
+                                             use_logits=True,
+                                             metrics=pretraining_metrics)
+    elif model_type == 'simpletransformer':
+        pretraining_model = SimpleTransformerMaskedLM(vocab_size=tiny_bert_tokenizer.vocab_size)
+    else:
+        raise ValueError('Unsupported model type')
+    pretraining_output_path = pretraining_test_helper(tiny_bert_tokenizer, pretraining_model, algorithms, tmp_path,
+                                                      device)
+
+    # finetuning
+    if model_type == 'tinybert_hf':
+        finetuning_metric = Accuracy()
+        hf_finetuning_model, _ = HuggingFaceModel.hf_from_composer_checkpoint(
+            pretraining_output_path,
+            model_instantiation_class='transformers.AutoModelForSequenceClassification',
+            model_config_kwargs={'num_labels': 3})
+        finetuning_model = HuggingFaceModel(model=hf_finetuning_model,
+                                            tokenizer=tiny_bert_tokenizer,
+                                            use_logits=True,
+                                            metrics=[finetuning_metric])
+    elif model_type == 'simpletransformer':
+        finetuning_model = SimpleTransformerClassifier(vocab_size=tiny_bert_tokenizer.vocab_size, num_classes=3)
+    else:
+        raise ValueError('Unsupported model type.')
+    finetuning_trainer, finetuning_dataloader = finetuning_test_helper(tiny_bert_tokenizer, finetuning_model,
+                                                                       algorithms, pretraining_output_path, tmp_path,
+                                                                       device)
+
+    # inference
+    batch = next(iter(finetuning_dataloader))
+    finetuning_trainer.state.model.to('cpu')
+    finetuning_trainer.state.model.eval()
```

Comment:
If this is the full NLP pipeline, we want to try saving the checkpoint to a bucket etc. and then loading from the bucket back for inference?

Reply:
:+1: